src/formatDate.js

```diff
@@ -8,7 +8,34 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separator = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(separator);
+  const dateComponents = {};
+  const newSeparator = toFormat[toFormat.length - 1];
+
+  fromFormat.forEach((el, index) => {
+    if (el !== separator) {
+      dateComponents[el] = dateParts[index];
+    }
+  });
+
+  if (dateComponents['YY']) {
+    dateComponents['YYYY'] =
+      dateComponents['YY'] < 30
+        ? '20' + dateComponents['YY']
+        : '19' + dateComponents['YY'];
+  }
+
+  if (dateComponents['YYYY'] && toFormat.includes('YY')) {
+    dateComponents['YY'] = dateComponents['YYYY'].slice(-2);
```

Comment:
Similar to the previous comment, ensure that 'YYYY' contains a 4-digit year before slicing the last two digits. Unexpected input could lead to incorrect results.